From 9afe4cfe6b5ec04bf07f06d36ee5ae34833e1dcd Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Mon, 20 Dec 2021 06:41:20 +0200
Subject: [PATCH] Revert "[SW-63186] gaudi2: sec: set axprot for arc farm"

As per alignment with LKD, MSTR_IF's OVRD register should not be used
to set to "USER level".

This is because sub-blocks which should be secure won’t be able to work,
as MSTR_IF's OVRD bit takes precedence over the engine's internal PROT
configuration.

This reverts commit 3534a0fbb2657e0260c30129f5211131e2240315.

Change-Id: Ieb9a691c2b77900e96706f440d8be9175c3c3596
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 8d04a90..3589fcc 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -36,8 +36,6 @@
 #include <rot0_mstr_if_core_hbw_regs.h>
 #include <rot0_mstr_if_core_lbw_regs.h>
 #include <pmmu_hbw_mstr_if_core_hbw_regs.h>
-#include <arc_farm_kdma_mstr_if_core_hbw_regs.h>
-#include <arc_farm_kdma_mstr_if_core_lbw_regs.h>
 
 /*
  * instances_num : Number of instances of RR to program.
@@ -954,14 +952,6 @@
 	regmap_writel(AXPROT_SEC_VAL, mmPMMU_HBW_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
-static void preboot_secure_arc_farm(void)
-{
-	/* Set arc farm to secure level */
-	regmap_writel(AXPROT_SEC_VAL, mmARC_FARM_KDMA_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-
-	regmap_writel(AXPROT_SEC_VAL, mmARC_FARM_KDMA_MSTR_IF_CORE_LBW_AXPROT_OVRD);
-}
-
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
@@ -972,7 +962,6 @@
 		hl_dev_prot_bits_init();
 		preboot_secure_rot();
 		preboot_secure_pmmu();
-		preboot_secure_arc_farm();
 		preboot_secure_lbw();
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
